packages/app/src/cli/services/app-logs/processes/polling-app-logs.ts

```diff
@@ -0,0 +1,161 @@
+import {ExtensionSpecification} from '../../../models/extensions/specification.js'
+import {DeveloperPlatformClient} from '../../../utilities/developer-platform-client.js'
+import {outputWarn, outputDebug} from '@shopify/cli-kit/node/output'
+import {partnersFqdn} from '@shopify/cli-kit/node/context/fqdn'
+
+export interface AppEventData {
+  shop_id: number
+  api_client_id: number
+  payload: string
+  event_type: string
+  source: string
+  source_namespace: string
+  cursor: string
+  status: 'success' | 'failure'
+  log_timestamp: string
+}
+
+export interface AppEventData {
+  shop_id: number
+  api_client_id: number
+  payload: string
+  event_type: string
+  source: string
+  source_namespace: string
+  cursor: string
+  status: 'success' | 'failure'
+  log_timestamp: string
+}
+
+export interface LogsOptions {
+  apiKey?: string
+  storeFqdn?: string
+  path?: string
+  source?: string
+  status?: string
+  configName?: string
+  directory: string
+  userProvidedConfigName?: string
+  specifications?: ExtensionSpecification[]
+  remoteFlags?: Flag[]
+  reset: boolean
+}
+
+export interface SubscribeOptions {
+  developerPlatformClient: DeveloperPlatformClient
+  storeId: string
+  apiKey: string
+}
+
+enum Flag {
+  DeclarativeWebhooks,
+}
+
+export const subscribeProcess = async ({storeId, apiKey, developerPlatformClient}: SubscribeOptions) => {
+  const appLogsSubscribeVariables = {
+    shopIds: [storeId],
+    apiKey,
+    token: '',
+  }
+  const result = await developerPlatformClient.subscribeToAppLogs(appLogsSubscribeVariables)
+  const {jwtToken, success, errors} = result.appLogsSubscribe
+  outputDebug(`Token: ${jwtToken}\n`)
+  outputDebug(`API Key: ${appLogsSubscribeVariables.apiKey}\n`)
+  if (errors && errors.length > 0) {
+    outputWarn(`Errors subscribing to app logs: ${errors.join(', ')}`)
+    outputWarn('App log streaming is not available in this `log` session.')
+    return
```

Comment:
For this sort of thing, throw an appropriate error instead. We'll then know whether to treat this as a bug or user error.  And there's a standard format for failures. The two to use are:
- `AbortError` -- user error, i.e. not Shopify's fault
- `BugError` -- a bug, something that's our fault

I don't know the situation that causes an error here and hence which is correct.